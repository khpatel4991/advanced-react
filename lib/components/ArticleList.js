import React from 'react';
import Article from 'components/Article';

const ArticleList = ({ articles }) => {
  return (
    <div>
      {Object.values(articles).map((article) => (
        <Article key={article.id} article={article} />
      ))}
    </div>
  );
};

export default ArticleList;
