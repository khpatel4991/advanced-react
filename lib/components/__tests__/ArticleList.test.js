import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import ArticleList from 'components/ArticleList';

configure({ adapter: new Adapter() });

describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: { id: 'a', title: 't1', body: 'b1', date: 'd1' },
      b: { id: 'b', title: 't2', body: 'b2', date: 'd2' }
    }
  };

  it('renders correctly', () => {
    const wrapper = shallow(<ArticleList {...testProps} />);
    expect(wrapper.find('ArticleContainer').length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});
