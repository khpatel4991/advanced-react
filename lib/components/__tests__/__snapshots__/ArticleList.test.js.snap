// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "a": Object {
          "body": "b1",
          "date": "d1",
          "id": "a",
          "title": "t1",
        },
        "b": Object {
          "body": "b2",
          "date": "d2",
          "id": "b",
          "title": "t2",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ArticleContainer
          article={
            Object {
              "body": "b1",
              "date": "d1",
              "id": "a",
              "title": "t1",
            }
          }
        />,
        <ArticleContainer
          article={
            Object {
              "body": "b2",
              "date": "d2",
              "id": "b",
              "title": "t2",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "a",
        "nodeType": "class",
        "props": Object {
          "article": Object {
            "body": "b1",
            "date": "d1",
            "id": "a",
            "title": "t1",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "b",
        "nodeType": "class",
        "props": Object {
          "article": Object {
            "body": "b2",
            "date": "d2",
            "id": "b",
            "title": "t2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ArticleContainer
            article={
              Object {
                "body": "b1",
                "date": "d1",
                "id": "a",
                "title": "t1",
              }
            }
          />,
          <ArticleContainer
            article={
              Object {
                "body": "b2",
                "date": "d2",
                "id": "b",
                "title": "t2",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "a",
          "nodeType": "class",
          "props": Object {
            "article": Object {
              "body": "b1",
              "date": "d1",
              "id": "a",
              "title": "t1",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "b",
          "nodeType": "class",
          "props": Object {
            "article": Object {
              "body": "b2",
              "date": "d2",
              "id": "b",
              "title": "t2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
